scripts/dev.sh

```
@@ -281,13 +295,51 @@ start_modular_server() {
     fi
     
     # Clear the log file and add a startup marker to capture fresh logs
-    echo "=== Server Starting at $(date) ===" > "$LOG_FILE"
+    # Rotate logs if too large (e.g., >10MB) or just simple rotation
+    if [[ -f "$LOG_FILE" ]]; then
+        local fsize=$(stat -f%z "$LOG_FILE" 2>/dev/null || echo 0)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)